tests/cpptests/test_cpp_query.cpp

```diff
@@ -128,6 +136,19 @@ TEST_F(QueryTest, testParser_delta) {
   assertInvalidQuery_v(1, "*=>[KNN $K @vec_field $BLOB EF $ef foo bar x 5 AS score]");
   assertInvalidQuery_v(1, "*=>[KNN $K @vec_field $BLOB foo bar x 5]");
 
+  // From version 5, NEGATION can't be used between the colon and the term
+  assertValidQuery_v(1, "@foo:-as");
+  assertValidQuery_v(2, "@foo:-as");
+  assertInvalidQuery_v(5, "@foo:-as");
+  assertValidQuery_v(1, "-@foo:as");
+  assertValidQuery_v(2, "-@foo:as");
+  assertValidQuery_v(5, "-@foo:as");
+
+  // From version 5, punctuation chars are not ignored, they should be escaped
+  assertValidQuery_v(1,"hello world&good");
+  assertValidQuery_v(2,"hello world&good");
+  assertInvalidQuery_v(5,"hello world&good");
```

Comment:
Test it is valid when escaping the punctuation

Reply:
OK. Test included.